megamek/src/megamek/common/enums/SkillLevel.java

```diff
@@ -0,0 +1,165 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.common.enums;
+
+import megamek.MegaMek;
+import megamek.common.util.EncodeControl;
+
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum SkillLevel {
+    //region Enum Declarations
+    NONE("SkillLevel.NONE.text", "SkillLevel.NONE.toolTipText"),
+    ULTRA_GREEN("SkillLevel.ULTRA_GREEN.text", "SkillLevel.ULTRA_GREEN.toolTipText"),
+    GREEN("SkillLevel.GREEN.text", "SkillLevel.GREEN.toolTipText"),
+    REGULAR("SkillLevel.REGULAR.text", "SkillLevel.REGULAR.toolTipText"),
+    VETERAN("SkillLevel.VETERAN.text", "SkillLevel.VETERAN.toolTipText"),
+    ELITE("SkillLevel.ELITE.text", "SkillLevel.ELITE.toolTipText"),
+    HEROIC("SkillLevel.HEROIC.text", "SkillLevel.HEROIC.toolTipText");
+    //endregion Enum Declarations
+
+    //region Variable Declarations
+    private final String name;
+    private final String toolTipText;
+    //endregion Variable Declarations
+
+    //region Constructors
+    SkillLevel(final String name, final String toolTipText) {
+        final ResourceBundle resources = ResourceBundle.getBundle("megamek.common.messages", new EncodeControl());
+        this.name = resources.getString(name);
+        this.toolTipText = resources.getString(toolTipText);
+    }
+    //endregion Constructors
+
+    //region Getters
+    public String getToolTipText() {
+        return toolTipText;
+    }
+    //endregion Getters
+
+    //region Boolean Comparisons
+    public boolean isNone() {
+        return this == NONE;
+    }
+
+    public boolean isUltraGreen() {
+        return this == ULTRA_GREEN;
+    }
+
+    public boolean isGreen() {
+        return this == GREEN;
+    }
+
+    public boolean isRegular() {
+        return this == REGULAR;
+    }
+
+    public boolean isVeteran() {
+        return this == VETERAN;
+    }
+
+    public boolean isElite() {
+        return this == ELITE;
+    }
+
+    public boolean isHeroic() {
+        return this == HEROIC;
+    }
+
+    public boolean isVeteranOrGreater() {
+        return isVeteran() || isEliteOrGreater();
+    }
+
+    public boolean isEliteOrGreater() {
+        return isElite() || isHeroic();
+    }
+    //endregion Boolean Comparisons
+
+    /**
+     * This returns the default skill values by level. This should never return the value for NONE,
+     * as NONE means one does not have the skill.
+     * @return the default skill array pairing
+     */
+    public int[] getDefaultSkillValues() {
+        switch (this) {
+            case NONE:
+                MegaMek.getLogger().error("Attempting to get illegal default skill values for NONE Skill Level. Returning { 10, 10 }");
+                return new int[]{ 10, 10 };
```

Comment:
Will 10,10 cause errors within the UI since the actual "maximum" is 8?

Reply:
Good point, and one I'd rather not test out. Swapped to 8.